statsd/buffer.go

```diff
@@ -4,11 +4,15 @@ import (
 	"strconv"
 )
 
-type bufferFullError string
+// MessageTooLongError is an error returned when a sample, event or service check is too large once serialized. See
+// WithMaxBytesPerPayload option for more details.
+type MessageTooLongError struct{}
 
-func (e bufferFullError) Error() string { return string(e) }
+func (e MessageTooLongError) Error() string {
+	return "message too long. See 'WithMaxBytesPerPayload' documentation."
+}
 
-const errBufferFull = bufferFullError("statsd buffer is full")
+var errBufferFull = &MessageTooLongError{}
```

Comment:
Doesn't need to be a pointer.

Reply:
Since we return this error often I wanted to avoid a copy each time.

Reply:
`MessageTooLongError` is an empty struct so its size is `0`.

Reply:
Indeed ! Forgot we move the error string :+1: